nomos-libp2p/src/behaviour/nat/address_mapper/protocols/pcp/mapping.rs

```diff
@@ -0,0 +1,338 @@
+use std::{net::Ipv4Addr, num::NonZeroU16};
+
+use zerocopy::FromBytes as _;
+
+use crate::behaviour::nat::address_mapper::protocols::pcp::{
+    client::PcpError,
+    wire::{Ipv4MappedIpv6, PcpMapResponse, Protocol, ResultCode, PCP_MAP_SIZE},
+};
+
+#[derive(Debug)]
+pub struct Mapping {
+    #[cfg_attr(not(test), expect(dead_code, reason = "Part of protocol"))]
```

Comment:
There is a lint that we have not enabled (yet), that will discourage using `#[cfg(not(test))]` for the reasons explained in the Clippy lint: https://rust-lang.github.io/rust-clippy/master/index.html#cfg_not_test. I believe this would trigger that lint as well. Could this be worked around in a different way than marking fields as non-test only? For instance, if the fields are only accessed in tests, then they could all be marked as private, and we could expose public getters that would be called in tests to check the relevant conditions.